WebXR Integration 🥽

cvachha opened the pull request:
- Adding initial implementation for WebXR support. Currently this supports a basic viewer and some controller based locomotion. 
- Branched from brent/splatting

 Currently work in progress features
1. Scene navigation (controllers + hand tracking)
2. Floating UI panels
3. Support for VR input for interactive clickable events

This supports VR and AR (mixed reality) mode

https://github.com/nerfstudio-project/viser/assets/9502341/807532a3-fb18-4bed-a957-d94aa357fabe

cvachha committed:
Initial WebXR integration

Adding initial implementation for WebXR support. Currently this supports a basic viewer and some controller based locomotion. Branched from brent/splatting

brentyi commented:
@cvachha I merged some of the experimental branches into `main` to make life easier for us. To clean up the commit history I then rebased your WebXR commit, do you mind double-checking that I didn't break anything?

If you've made other local changes this might cause problems, you could commit them somewhere and then cherry-pick back into this branch. Sorry 😬

cvachha committed:
Updating to react-xr in progress

In progress updating from react-xr v5 to v6, but running into errors in VR mode.
